Utils/PluginActivator/source/COMRPCStarter.cpp

```diff
@@ -0,0 +1,149 @@
+/*
+ * If not stated otherwise in this file or this component's LICENSE file the
+ * following copyright and licenses apply:
+ *
+ * Copyright 2020 Metrological
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "COMRPCStarter.h"
+
+#include "Log.h"
+
+#include <chrono>
+#include <thread>
+
+COMRPCStarter::COMRPCStarter(const string& pluginName)
+    : mPluginName(pluginName)
+    , mEngine(Core::ProxyType<RPC::InvokeServerType<1, 0, 4>>::Create())
+    , mClient(Core::ProxyType<RPC::CommunicatorClient>::Create(getConnectionEndpoint(),
+          Core::ProxyType<Core::IIPCServer>(mEngine)))
+{
+    // Announce our arrival
+    mEngine->Announcements(mClient->Announcement());
+
+    if (!mClient.IsValid()) {
+        LOG_ERROR(pluginName.c_str(), "Failed to create valid COM-RPC client");
```

Comment:
Thunder has an internal concept of LOGGING and TRACING with a different scope (logging -> operator/tester, always on, tracing ->development always off). This should waorking with Thunder and working with its plugins give a consistent look and feel. We will address this topic next week as well.